web/client/utils/FeatureGridUtils.js

```diff
@@ -162,19 +162,50 @@ module.exports = {
     applyAllChanges: (orig, changes = {}) => applyChanges(orig, changes[orig.id] || {}),
     applyChanges,
     gridUpdateToQueryUpdate: ({attribute, operator, value, type} = {}, oldFilterObj = {}) => {
+        const andGroup = {
+            id: 9999, // dummy number representing group id
+            logic: "AND",
+            index: 0
+        };
+
+        let mergeFilters = oldFilterObj.groupFields 
+            ? (oldFilterObj.groupFields.filter(obj => obj.id === 1 ).length > 0 && oldFilterObj.filterFields.length > 0 
+               ? true // checks if query panel filter is active
+                : oldFilterObj.filterFields.length - 1 > 0 
+                    || ( oldFilterObj.filterFields[0] && oldFilterObj.filterFields[0].hasOwnProperty('loading'))) 
+            : false;
+        if ( mergeFilters ) {
+            let multipleFields = oldFilterObj.filterFields.length > 2; // checks for multiple fields in query panel filter
+
+            let {filterFields: oldFilterFields, groupFields: oldGroupFields } = oldFilterObj;
+            let mappedFilterFields = oldFilterFields.map(obj => obj.groupId === 1 && !multipleFields ? {...obj, groupId: andGroup.id} : obj);
+            let slicedFilterFields = mappedFilterFields.filter(obj => obj.hasOwnProperty('loading'));
```

Comment:
Why special fields with loading

Reply:
Only filter objects coming from query panel filter have property loading, I have therefore used property loading to remove filter objects coming from Featuregrid filter, and keep only the ones coming from query panel filter.